utils/s2n_mem.c

```diff
@@ -14,7 +14,7 @@
  */
 
 #define  _DEFAULT_SOURCE 1
-#if !defined(__APPLE__) && !defined(__FreeBSD__)
+#if !defined(__APPLE__) && !defined(__FreeBSD__) && !defined(__NetBSD__)
```

Comment:
Created an issue for a more durable solution: https://github.com/aws/s2n-tls/issues/3132